Turn Order and Combat Action

SiogunJH opened the pull request:
Added TextMeshPro package.
Added Sample ShaderGraph Shaders package.

Implemented two Combat Actions:
Pistol Shot
Move

Added tile highlight for available target tiles and selected tiled.

SiogunJH committed:
TMPro package

SiogunJH committed:
Ability const fix

SiogunJH committed:
TurnManager renamed to CombatManager

SiogunJH committed:
Tiles Filtering

SiogunJH committed:
Update GridTileController.cs

SiogunJH committed:
CombatAbilitiesUIManager

SiogunJH committed:
Update Character.cs

SiogunJH committed:
Update CombatManager.cs

SiogunJH committed:
UI prefabs for Combat Abilities display

SiogunJH committed:
Update CombatManager.cs

SiogunJH committed:
Update CombatAbilityUIManager.cs

SiogunJH committed:
Update BattleTestMap.unity

SiogunJH committed:
Starting combat will enable and clear Combat Ability UI

SiogunJH committed:
Sample import for Shader Graph

SiogunJH committed:
Shader and Material for tiles

SiogunJH committed:
Tile prefab update

SiogunJH committed:
Highlight logic

SiogunJH committed:
Handlers EffectTyp fix

SiogunJH committed:
OnSelect/Deselect events and ability button creation

SiogunJH committed:
Ability finish logic

SiogunJH committed:
TryToEndTurn and RemoveActionPoint logic

SiogunJH committed:
CombatAbilitiesPanel prefab

SiogunJH committed:
Scene update

SiogunJH committed:
Reduced the default amount of ability points to 2

SiogunJH committed:
Ability: Kill -> Pistol Shot

SiogunJH committed:
Ability: Move update

Move can now target adjacent tiles only

SiogunJH committed:
Initialize Combat and Grid on startup

SiogunJH committed:
Ability name wrapping

SiogunJH committed:
Enemy prefab update

SiogunJH committed:
Move Handler

SiogunJH committed:
Debug log for abilities and effects

SiogunJH committed:
Modified GridEffectHandler logic

Now, instead of Character-based effect distribution, GFH will try to execute abilities on each tile and their respective occupants. This is to allow DoEffect to perform effect even when no occupant are targeted (for abilities such as Move)

SiogunJH committed:
DoEffect handlers update

SiogunJH committed:
Additional Assert in ExecuteAbility

SiogunJH committed:
Safety check for Damage handler

SiogunJH committed:
Update MoveHandler.cs

SiogunJH committed:
ClearTileHighlight fix

SiogunJH committed:
Update DamageKineticHandler.cs

SiogunJH committed:
Expose CurrentHealth from Character

SiogunJH committed:
Log spawned enemies

SiogunJH committed:
Log damage dealt

SiogunJH committed:
Increase PistolShot damage to 3

SiogunJH committed:
Prefabs update

SiogunJH committed:
Receiving damage and dying logic

AbexCorp commented on Assets/Scripts/Characters/Character.cs:
```diff
@@ -248,12 +258,18 @@ private void ResetAttributeChanges()
     [SerializeField, HideInInspector] private bool _hadTurn = false;
     public bool HadTurn => _hadTurn;
 
+    public void RemoveActionPoints(int amount)
```
Shouldn't this first check if there is enough points to remove?

SiogunJH committed:
RemoveActionPoint safety validation

Removing action poins should be done after performing an action or after confirming it, so logic related to verifying if a Combatant has enough AP to perform an Action, should be placed there.

This is just a safety check, to ensure that logic (when implemented) will work fine

SiogunJH committed:
Death logic revamped

Added a wrapper to logic responsible for checking Death requirements, in case we would need to bypass them or branch logic based on its outcome.

Death logic now unregisters dying Combatant from a Tile it occupies.

Removed the FUCKASS comment.
